lib/active_storage/service/bunny_service.rb

```diff
@@ -25,10 +28,16 @@ def upload(key, io, checksum: nil, filename: nil, content_type: nil, disposition
     end
 
     def download(key, &block)
-      instrument :download, key: key do
-        object_for(key).get_file.to_s.force_encoding(Encoding::BINARY)
-      rescue StandardError
-        raise ActiveStorage::FileNotFoundError
+      if block_given?
+        instrument :streaming_download, { key: key } do
+          stream(key, &block)
+        end
+      else
+        instrument :download, key: key do
+          io = StringIO.new object_for(key).get_file
+
+          io
```

Comment:
```suggestion
          io.set_encoding(Encoding::BINARY)
```

Reply:
better to move this to another private method since it is being used again in stream.